include/ensmallen_bits/moead/moead.hpp

```diff
@@ -18,35 +16,38 @@
 #ifndef ENSMALLEN_MOEAD_MOEAD_HPP
 #define ENSMALLEN_MOEAD_MOEAD_HPP
 
+//! Decomposition policies
+#include "decomposition_policies/tchebycheff_decomposition.hpp"
+#include "decomposition_policies/weighted_decomposition.hpp"
+#include "decomposition_policies/pbi_decomposition.hpp"
+
+//! Weight Init policies
+#include "weight_init_policies/bbs_init.hpp"
 namespace ens {
 
 /**
- * This class implements the MOEA/D algorithm with Differential Evolution
- * crossover. Step numbers used in different parts of the implementation
- * correspond to the step number used in the original algorithm by the author.
+ * MOEA/D-DE (Multi Objective Evolutionary Algorithm based on Decompositon - 
+ * Differential Variant) is a multiobjective optimization algorithm. This class 
+ * implements the said optimizer. 
+ *
+ * The algorithm works by generating a candidate population from a fixed starting point. 
+ * Reference directions are generated to guide the optimization process towards the Pareto Front. 
+ * Further, a decomposition function is defined to decompose the problem to a scalar optimization 
+ * objective. Utilizing genetic operators, offsprings are generated with better decomposition values 
+ * to replace the neighboring parent solutions. 
  *
  * For more information, see the following:
  * @code
- * @article{article,
- * author = {Zhang, Qingfu and Li, Hui},
- * year = {2008},
- * pages = {712 - 731},
- * title = {MOEA/D: A Multiobjective Evolutionary Algorithm Based on
- *    Decomposition},
- * journal = {Evolutionary Computation, IEEE Transactions on},
- *
- * @article{4633340,
- * author={H. {Li} and Q. {Zhang}},
- * year={2009},
- * pages={284-302},}
- * title={Multiobjective Optimization Problems With Complicated Pareto Sets, MOEA/D and NSGA-II},
- * journal={IEEE Transactions on Evolutionary Computation},
+ * @article{li2008multiobjective,
+ * title={Multiobjective optimization problems with complicated Pareto sets, MOEA/D and NSGA-II},
+ * author={Li, Hui and Zhang, Qingfu},
+ * journal={IEEE transactions on evolutionary computation},
+ * pages={284--302},
+ * year={2008},
  * @endcode
- *
- * MOEA/D can optimize arbitrary multi-objective functions. For more details,
- * see the documentation on function types included with this distribution or
- * on the ensmallen website.
  */
+template<typename InitPolicyType = BayesianBootstrap,
+         typename DecompPolicyType = Tchebycheff>
```

Comment:
I think we can be a little bit more verbose on the `DecompPolicyType` naming here, `DecompositionPolicyType`?

Reply:
Can we keep decomp policy? I have commented what it means. The problem is it gets too verbose and takes up lots of space. 

Let me know if you think otherwise.

Reply:
Okay.